extensions/parquet/table/src/test/java/io/deephaven/parquet/table/ParquetTableReadWriteTest.java

```diff
@@ -614,6 +614,94 @@ private Table maybeFixBigDecimal(Table toFix) {
                 .dropColumns("bdColE");
     }
 
+    /**
+     * Test if the current code can read the parquet data written by the old code. There is logic in
+     * {@link ColumnChunkPageStore#create} that decides page store based on the version of the parquet file. The old
+     * data is generated using following logic:
+     * 
+     * <pre>
+     *  // Enforce a smaller page size to write multiple pages
+     *  final ParquetInstructions writeInstructions = new ParquetInstructions.Builder()
+     *        .setTargetPageSize(ParquetInstructions.MIN_TARGET_PAGE_SIZE)
+     *        .build();
+     *
+     *  final Table table = getTableFlat(5000, true, false);
+     *  ParquetTools.writeTable(table, new File("ReferenceParquetData.parquet"), writeInstructions);
+     *
+     *  Table vectorTable = table.groupBy().select();
+     *  vectorTable = vectorTable.join(TableTools.emptyTable(100)).select();
+     *  ParquetTools.writeTable(vectorTable, new File("ReferenceParquetVectorData.parquet"), writeInstructions);
+     *
+     *  final Table arrayTable = vectorTable.updateView(vectorTable.getColumnSourceMap().keySet().stream()
+     *         .map(name -> name + " = " + name + ".toArray()")
+     *         .toArray(String[]::new));
+     *  ParquetTools.writeTable(arrayTable, new File("ReferenceParquetArrayData.parquet"), writeInstructions);
+     * </pre>
+     */
+    @Test
+    public void testReadOldParquetData() {
+        String path = ParquetTableReadWriteTest.class.getResource("/ReferenceParquetData.parquet").getFile();
+        try {
+            ParquetTools.readTable(new File(path)).select();
+        } catch (RuntimeException e) {
+            if (e.getCause() instanceof InvalidParquetFileException) {
+                final String InvalidParquetFileErrorMsgString = "Invalid parquet file detected, please ensure the " +
+                        "file is fetched properly from Git LFS. Run commands 'git lfs install; git lfs pull' inside " +
+                        "the repo to pull the files from LFS. Check cause of exception for more details.";
+                throw new UncheckedDeephavenException(InvalidParquetFileErrorMsgString, e.getCause());
+            }
+            throw e;
+        }
+        final ParquetMetadata metadata = new ParquetTableLocationKey(new File(path), 0, null).getMetadata();
+        assertTrue(metadata.getFileMetaData().getKeyValueMetaData().get("deephaven").contains("\"version\":\"0.4.0\""));
+
+        path = ParquetTableReadWriteTest.class.getResource("/ReferenceParquetVectorData.parquet").getFile();
+        ParquetTools.readTable(new File(path)).select();
```

Comment:
Given it's easy to generate the source data, I think we should check that the table contents are equal - this will also save you from needing to materialize via select.

Going to put a plug in for `readSingleFileTable` once #4831 merges

Ditto for other call site like this.

Reply:
I didn't want to generate the data because if we change the method `getTableFlat` in future (which we do change very frequently as we add new types), these tests will start to fail. 

Also, currently, `getTablFlat` returns a column `"someTime = DateTimeUtils.now() + i"`
So we cannot regenerate that data because of the `now`.   This is still fixable since I can change the column. But in general didn't like the overall thing.